mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/components/SlippageTolerance.tsx

```diff
@@ -1,124 +1,211 @@
-      <View style={tailwind('w-6/12 flex-grow')}>
+    <>
+      <View style={tailwind('flex-row mb-3 flex-wrap')}>
+        {percentageList.map((value) => (
+          <SlippageButton
+            key={value}
+            onPress={() => {
+              setIsCustomSlippage(false)
+              onSlippageChange(value)
+            }}
+            isActive={!isCustomSlippage && selectedSlippage === value}
+            label={`${value}%`}
+          />
+        ))}
+        <SlippageButton
+          onPress={() => {
+            setIsCustomSlippage(true)
+          }}
+          icon={
+            <ThemedIcon
+              size={14}
+              name='edit'
+              iconType='MaterialIcons'
+              style={tailwind('mr-1')}
+              dark={tailwind(`${isCustomSlippage ? 'text-darkprimary-500' : 'text-gray-50'}`)}
+              light={tailwind(`${isCustomSlippage ? 'text-primary-500' : 'text-gray-900'}`)}
+            />
+          }
+          isActive={isCustomSlippage}
+          label={translate('screens/SlippageTolerance', 'Custom')}
+        />
+      </View>
+
+      {isCustomSlippage && (
+        <ThemedView
+          dark={tailwind('bg-transparent')}
+          light={tailwind('bg-transparent')}
+          style={tailwind('flex-row w-full')}
+        >
+          <WalletTextInput
+            autoFocus
+            onChangeText={onSlippageChange}
+            keyboardType='numeric'
+            autoCapitalize='none'
+            placeholder='0.00%'
+            style={tailwind('flex-grow h-8')}
+            containerStyle='mb-1 w-full flex-col'
+            testID='slippage_input'
+            value={selectedSlippage !== undefined ? selectedSlippage.toString() : ''}
+            displayClearButton={selectedSlippage !== undefined}
+            onClearButtonPress={() => onSlippageChange('')}
+            inputType='numeric'
+            valid={isSlippageValid()}
+            inlineText={slippageError}
+          />
+        </ThemedView>
+      )}
+
+      {isRiskWarningDisplayed && (
         <ThemedText
-          dark={tailwind('text-gray-200')}
-          light={tailwind('text-black')}
-          style={tailwind('text-sm mr-1')}
-          testID='slippage_title'
+          light={tailwind('text-error-500')}
+          dark={tailwind('text-darkerror-500')}
+          style={tailwind('text-sm mb-3')}
+          testID='slippage_warning'
         >
-          {translate('screens/SlippageTolerance', 'Slippage tolerance')}
+          {translate('screens/SlippageTolerance', 'Proceed at your own risk')}
         </ThemedText>
-      </View>
-      <View
-        style={tailwind('flex w-6/12 flex-row justify-end flex-wrap items-center')}
+      )}
+    </>
+  )
+}
+
+function SlippageButton ({ onPress, isActive, label, icon }: { onPress: () => void, isActive: boolean, label: string, icon?: ReactElement }): JSX.Element {
+  const buttonStyles = 'flex flex-row px-2 py-1.5 border border-gray-300 rounded items-center'
```

Comment:
Can we add a different border color on dark mode? 

On light mode, the button border is consistent with other input field, but on dark mode, it is taking more attention than the other